src/Documents/RQL/QueryBuilder.ts

```diff
@@ -0,0 +1,164 @@
+    this._where = '';
+    this._order = '';
+    this._or = true;
+    this._fromCollection = false;
+    this.defaultOperator = '';
+    this.operator = this.defaultOperator;
+    this.negate = false;
+    this.rqlText = null;
+  }
+
+  public get and() {
+    this.operator = QueryOperators.AND;
+    return this;
+  }
+
+  public get or() {
+    this.operator = QueryOperators.OR;
+    return this;
+  }
+
+  public get not() {
+    this.negate = true;
+    return this;
+  }
+
+  public select(fields) {
+
+    if (TypeUtil.isArray(fields)) {
+      this._select = fields.join(', ');
+    }
+    else if (fields) {
+      this._select = fields;
+    }
+    else {
+      this._select = '*';
+    }
+
+    return this;
+
+  }
+
+  public from(collectionOrIndex, fromCollection) {
+
+    if(fromCollection) {
+      this._fromCollection = collectionOrIndex;
+    }
+    else {
+      this._from = collectionOrIndex;
+    }
+
+    return this;
+
+  }
+
+  public order(field, direction = 'ASC') {
+
+    this._order = StringUtil.format(`{0} {1}`, field, direction);
+
+    return this;
+
+  }
+
+  public where(conditionType, conditionField, conditionValue = null) {
+
+    switch (conditionType) {
+      case rqlOperators.greaterThan:
+        this.rqlText = StringUtil.format(`{0}>{1}`, conditionField, conditionValue);
+        break;
+      case rqlOperators.lessThan:
+        this.rqlText = StringUtil.format(`{0}<{1}`, conditionField, conditionValue);
+        break;
+      case rqlOperators.equals:
+        (conditionValue === null || conditionValue === 'null') ?
+          this.rqlText = StringUtil.format(`{0}={1} `, conditionField, conditionValue) :
+          this.rqlText = StringUtil.format(`{0}='{1}' `, conditionField, conditionValue);
+        break;
+      case rqlOperators.between:
+        this.rqlText = StringUtil.format(`{0} BETWEEN {1} AND {2}`, conditionField, conditionValue.start, conditionValue.end);
+        break;
+      case rqlOperators.startsWith:
+        this.rqlText = StringUtil.format(`StartsWith({0}, '{1}')`, conditionField, conditionValue);
+        break;
+      case rqlOperators.endsWith:
+        this.rqlText = StringUtil.format(`EndsWith({0}, '{1}')`, conditionField, conditionValue);
+        break;
+      case rqlOperators.in:
+        this.rqlText = StringUtil.format(`{0} IN ('{1}')`, conditionField, conditionValue);
+        break;
+      case rqlOperators.search:
+        this.rqlText = StringUtil.format(`search({0},'{1}') `, conditionField, conditionValue);
+        break;
+      case rqlOperators.boost:
+        this.rqlText = StringUtil.format(`boost({0} = '{1}', {2})`, conditionField.boostField, conditionField.boostValue, conditionValue);
+        break;
+    }
+
+    this.whereRaw(this.rqlText);
+
+    return this;
+
+  }
+
+
+  public whereRaw(rqlCondition) {
+
+    let addNot: string = this.negate ? 'NOT ' : '';
+
+    this._where += StringUtil.format(`{0} {1} {2}`, this.operator, addNot, rqlCondition);
```

Comment:
Consider using list of tokens
https://github.com/ravendb/ravendb/tree/v4.0/src/Raven.Client/Documents/Session/Tokens
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs#L113
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs#L1322-L1327